components/Modal/CreateImageUpload.js

```diff
@@ -0,0 +1,751 @@
+      "These changes will be committed before the image is created."
+  },
+  selectOne: {
+    defaultMessage: "Select one"
+  },
+  title: {
+    defaultMessage: "Create and Upload Image"
+  },
+  review: {
+    defaultMessage:
+      "Review the information below and click Finish to create the image and complete the tasks that were selected. "
+  }
+});
+
+class CreateImageUpload extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isOpen: false
+    };
+    this.open = this.open.bind(this);
+    this.close = this.close.bind(this);
+  }
+
+  open() {
+    this.setState({ isOpen: true });
+  }
+
+  close() {
+    this.setState({ isOpen: false });
+  }
+
+  render() {
+    return (
+      <React.Fragment>
+        <Button variant="secondary" onClick={this.open}>
+          <FormattedMessage defaultMessage="Create Image" />
+        </Button>
+        {this.state.isOpen && <CreateImageUploadModal {...this.props} close={this.close} isOpen={this.state.isOpen} />}
+      </React.Fragment>
+    );
+  }
+}
+
+class CreateImageUploadModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      imageType: "",
+      imageName: "",
+      showUploadAzureStep: false,
+      showUploadOpenstackStep: false,
+      showUploadVSphereStep: false,
+      showReviewStep: false,
+      uploadService: "",
+      uploadSettings: {}
+    };
+    this.isPendingChange = this.isPendingChange.bind(this);
+    this.setNotifications = this.setNotifications.bind(this);
+    this.setImageName = this.setImageName.bind(this);
+    this.setImageType = this.setImageType.bind(this);
+    this.setUploadSettings = this.setUploadSettings.bind(this);
+    this.setUploadService = this.setUploadService.bind(this);
+    this.handleCreateImage = this.handleCreateImage.bind(this);
+    this.handleCommit = this.handleCommit.bind(this);
+    this.handleStartCompose = this.handleStartCompose.bind(this);
+    this.handleNextStep = this.handleNextStep.bind(this);
+    this.handlePreviousStep = this.handlePreviousStep.bind(this);
+  }
+
+  componentWillMount() {
+    this.props.fetchingUploadProviders();
+    if (this.props.imageTypes.length === 0) {
+      this.props.fetchingComposeTypes();
+    }
+    if (this.props.composeQueueFetched === false) {
+      this.props.fetchingQueue();
+    }
+  }
+
+  componentWillUnmount() {
+    this.props.clearQueue();
+  }
+
+  setNotifications() {
+    this.props.layout.setNotifications();
+  }
+
+  setUploadService(_, event) {
+    this.setState({ uploadService: event.target.value });
+  }
+
+  setUploadSettings(value, event) {
+    const key = event.target.id;
+    this.setState(prevState => ({ uploadSettings: Object.assign({}, prevState.uploadSettings, { [key]: value }) }));
+  }
+
+  setImageType(imageType) {
+    this.setState({
+      imageType,
+      uploadService: ""
+    });
+  }
+
+  setImageName(imageName) {
```

Comment:
I am not fond of one liner functions. Just call setState directly instead of calling the function that calls setState.

Reply:
Fixed